base/uk.ac.stfc.isis.ibex.ui.journalviewer/src/uk/ac/stfc/isis/ibex/ui/journalviewer/SearchInput.java

```diff
@@ -0,0 +1,309 @@
+            public void widgetSelected(SelectionEvent e) {
+                dtToDate.setEnabled(chkTimeTo.getSelection());
+                dtToTime.setEnabled(chkTimeTo.getSelection());
+            }
+        });
+
+        cmpRBNumber = new Composite(cmpSearch, SWT.NONE);
+        cmpFilters.add(cmpRBNumber);
+        cmpRBNumber.setLayout(new GridLayout(1, false));
+
+        txtSearchRBNumber = new Text(cmpRBNumber, SWT.BORDER);
+        GridData gdTxtSearchRBNumber = new GridData(SWT.LEFT, SWT.CENTER, false, false, 1, 1);
+        gdTxtSearchRBNumber.widthHint = SEARCH_BOX_WIDTH;
+        txtSearchRBNumber.setLayoutData(gdTxtSearchRBNumber);
+
+        stackSearch.topControl = cmpRunNumber;
+
+        cmpUsers = new Composite(cmpSearch, SWT.NONE);
+        cmpFilters.add(cmpUsers);
+        cmpUsers.setLayout(new GridLayout(1, false));
+
+        txtSearchUsers = new Text(cmpUsers, SWT.BORDER);
+        GridData gdTxtSearchUsers = new GridData(SWT.LEFT, SWT.CENTER, false, false, 1, 1);
+        gdTxtSearchUsers.widthHint = SEARCH_BOX_WIDTH;
+        txtSearchUsers.setLayoutData(gdTxtSearchUsers);
+
+        cmpSearch.setTabList(new Control[] {cmpRunNumber, cmpTitle, cmpTimePicker, cmpRBNumber, cmpUsers});
+
+        chkNumberFrom.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                spinnerFromNumber.setEnabled(chkNumberFrom.getSelection());
+            }
+        });
+
+        chkNumberTo.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                spinnerToNumber.setEnabled(chkNumberTo.getSelection());
+            }
+        });
+
+        cmbFilterType.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                DISPLAY.asyncExec(new Runnable() {
+                    @Override
+                    public void run() {
+                        stackSearch.topControl = cmpFilters.get(cmbFilterType.getSelectionIndex());
+                        cmpSearch.layout();
+                    }
+                });
+            }
+        });
+    }
+
+    /**
+     * @return The current selection index of the filter type drop-down menu.
+     */
+    public int getCmbFilterTypeIndex() {
+        return cmbFilterType.getSelectionIndex();
+    }
+
+    /**
+     * @return The text in the search box of the currently active filter type.
+     */
+    public Optional<String> getActiveSearchText() {
+        if (stackSearch.topControl == cmpTitle) {
+            return Optional.of(txtSearchTitle.getText());
+        } else if (stackSearch.topControl == cmpRBNumber) {
+            return Optional.of(txtSearchRBNumber.getText());
+        } else if (stackSearch.topControl == cmpUsers) {
+            return Optional.of(txtSearchUsers.getText());
+        } else {
+            return null;
```

Comment:
This should presumably be `Optional.empty()` rather than null - otherwise we lose all the benefits of using an `Optional`

Reply:
Changed it to `Optional.empty()`.